drivers/openstack/openstack.go

```diff
@@ -736,7 +736,37 @@ func (d *Driver) waitForSSHServer() error {
 		"MachineId": d.MachineId,
 		"IP":        ip,
 	}).Debug("Waiting for the SSH server to be started...")
-	return ssh.WaitForTCP(fmt.Sprintf("%s:%d", ip, d.SSHPort))
+
+	err = ssh.WaitForTCP(fmt.Sprintf("%s:%d", ip, d.SSHPort))
```

Comment:
Or how about a new ssh.WaitForSSH function that would loop until "exit 0" could be successfully executed like @ggiamarchi implemented for the OpenStack driver? Other users of ssh.WaitForTCP could be changed to this as needed.